apps/condo/domains/news/schema/ExportNewsRecipientsService.js

```diff
@@ -0,0 +1,210 @@
+/**
+ * Generated by `createservice news.ExportNewsRecipientsService --type mutations`
+ */
+
+const dayjs = require('dayjs')
+const get = require('lodash/get')
+const isEmpty = require('lodash/isEmpty')
+const isNil = require('lodash/isNil')
+
+const conf = require('@open-condo/config')
+const { GQLError, GQLErrorCode: { INTERNAL_ERROR } } = require('@open-condo/keystone/errors')
+const { GQLCustomSchema } = require('@open-condo/keystone/schema')
+const { extractReqLocale } = require('@open-condo/locales/extractReqLocale')
+const { i18n } = require('@open-condo/locales/loader')
+
+const { NOT_FOUND } = require('@condo/domains/common/constants/errors')
+const { createExportFile } = require('@condo/domains/common/utils/createExportFile')
+const { getHeadersTranslations, EXPORT_TYPE_NEWS_RECIPIENTS } = require('@condo/domains/common/utils/exportToExcel')
+const { loadListByChunks } = require('@condo/domains/common/utils/serverSchema')
+const access = require('@condo/domains/news/access/ExportNewsRecipientsService')
+const { Property } = require('@condo/domains/property/utils/serverSchema')
+const { Resident } = require('@condo/domains/resident/utils/serverSchema')
+
+
+const getUnitsFromProperty = ({ map }) => (
+    map?.sections?.reduce((acc, section) => ([
+        ...acc,
+        ...getUnitsFromSection(section),
+    ]), []) || []
+)
+
+const getUnitsFromSection = (section) => section.floors.flatMap(floor => floor.units.map(unit => ({ unitName: unit.label, unitType: unit.unitType })))
+
+const buildExportFile = async ({ rows, locale }) => {
+    const YesMessage = i18n('Yes', { locale })
+    const NoMessage = i18n('No', { locale })
+    const HeaderMessage =  i18n('excelExport.sheetNames.newsRecipients', { locale })
+
+    const processedRows = rows.reduce((acc, row) => {
+        row.hasResident = row.hasResident ? YesMessage : NoMessage
+        return [...acc, row]
+    }, [])
+    const { url: linkToFile } = await createExportFile({
+        fileName: `news_recipients_${dayjs().format('DD_MM')}.xlsx`,
+        templatePath: './domains/news/templates/NewsRecipientsExportTemplate.xlsx',
+        replaces: {
+            header: HeaderMessage,
+            newsRecipients: processedRows,
+            i18n: {
+                ...getHeadersTranslations(EXPORT_TYPE_NEWS_RECIPIENTS, locale),
+                sheetName: i18n('excelExport.sheetNames.newsRecipients', { locale }),
+            },
+        },
+    })
+
+    return linkToFile
+}
+
+const ExportNewsRecipientsService = new GQLCustomSchema('ExportNewsRecipientsService', {
```

Comment:
Why not a background service pattern?

Reply:
Later it will be transferred to the worker task